src/signer/src/derivation_path.rs

```diff
@@ -0,0 +1,21 @@
+use candid::Principal;
+
+/// The schema, which is the first part of a derivation path.
+#[repr(u8)]
+pub enum Schema {
+    /// The derivation path for a principal, with no further domain separation.
+    ///
+    /// Please see `from_principal` for details.
+    Principal = 1,
+}
+
+impl From<Schema> for Vec<u8> {
+    fn from(s: Schema) -> Vec<u8> {
+        vec![s as u8]
+    }
+}
+
+/// A derivation path composed of: `vec![schema, principal]`.
+pub fn from_principal(p: &Principal) -> Vec<Vec<u8>> {
```

Comment:
This one is used for ETH, does it make sense to refer to ETH in the name of the method and schema?

Reply:
Good idea.  Done!